ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -141,6 +144,18 @@ public Topology getTopology() {
     return ImmutableList.of();
   }
 
+  public Map<LagInfoKey, Map<Integer, LagInfo>> getStoreToPartitionToLagMap() {
+    Map<LagInfoKey, Map<Integer, LagInfo>> storeToPartitionToLagMap = null;
+    try {
+      storeToPartitionToLagMap = kafkaStreams.allLocalStorePartitionLags().entrySet().stream()
+          .map(e -> Pair.of(LagInfoKey.of(getQueryApplicationId(), e.getKey()), e.getValue()))
```

Comment:
(I think you are handling this already). Note to self: When a new query is scheduled, it's possible that the sending node has the store created for it already i.e the aware of the query while the receiving node still may not have seen the queryID being scheduled on it.. So if we expect the receiving node to do any checks based on the queryApplicationId(), it may fail in a small window